modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/custom_fields.js

```diff
@@ -45,7 +45,7 @@ AUI.add(
 
 		var structureFieldIndexEnable = function() {
 			var indexTypeNode = A.one(
-				'#_' + Liferay.Portlet.list[0] + '_indexable'
+				'#_com_liferay_journal_web_portlet_JournalPortlet_indexable'
```

Comment:
Hey @knchau,

You can not hardcode a namespace, this is very fragile, if the name change the code will break, in addition you put the Journal, this cannot, since the DDM is a Framework that serves several portlets.

Try to using `A.all` instead of `A.one`, it will return an array.

Thanks!

Reply:
Hi @inacionery,

I see! I'm trying a different approach here: https://github.com/leoadb/liferay-portal/pull/575

I'm looping through the available namespaces until finding one that contains the hidden attribute. Let me know if you see any issues with this approach. Thank you!!